.gitlab-ci.yml

```diff
@@ -671,13 +673,16 @@ publish-draft-release:
 
 publish-to-crates-io:
   stage:                           publish
+  needs:
+    - job:                         unleash-check
```

Comment:
Either you should list the jobs necessary to be green in order to start this job, because this way it starts right after `unleash-check` without waiting anything else. Or remove this `needs` and the stage won't start if the previous jobs are not successful. I prefer the latter.

Reply:
Can we somehow make it so that it runs at `publish` but also make sure that `unleash-check` is never `allow_failure: true`?

Reply:
By `publish` I mean only after everything else completes from the `test` stage and so on

Reply:
We can run `unleash check` on the `build ` stage, so after all the code tests pass, and `unleash em-dragons` on the publish phase after everything's proven to be buildable.

Reply:
I can move this to the `build` stage if you want me to, running parallel to the tests is also good (at least wrt developer feedback latency, maybe not so much wrt CI builder overhead :grinning: )

Reply:
it's worth it, the test stage already brings a lot of concurrent jobs. [The](https://gitlab.parity.io/parity/substrate/-/pipelines/148586/builds) longest job in the test stage is `test-linux-stable `, which takes 36 minutes. This one takes ~32. The longest on the build stage is `check-polkadot-companion-build `, so unless we make use of caching (or simplify it somehow) it will severely contribute to the pipelines concurrency.